src/Tools/Source/RunTests/TestRunner.cs

```diff
@@ -156,29 +157,47 @@ string makeHelixWorkItemProject(AssemblyInfo assemblyInfo)
 
                 var commandLineArguments = _testExecutor.GetCommandLineArguments(assemblyInfo, useSingleQuotes: isUnix);
                 commandLineArguments = SecurityElement.Escape(commandLineArguments);
+                var setEnvironmentVariable = isUnix ? "export" : "set";
+
+                var command = new StringBuilder();
+                command.AppendLine(isUnix ? "ls -l" : "dir");
+                command.AppendLine(isUnix ? $"./rehydrate.sh" : $@"call .\rehydrate.cmd");
+                command.AppendLine(isUnix ? "ls -l" : "dir");
+                command.AppendLine($"{setEnvironmentVariable} DOTNET_ROLL_FORWARD=LatestMajor");
+                command.AppendLine($"{setEnvironmentVariable} DOTNET_ROLL_FORWARD_TO_PRERELEASE=1");
+                command.AppendLine("dotnet --info");
+
+                if (Environment.GetEnvironmentVariable("ROSLYN_TEST_IOPERATION") is string iop)
+                    command.AppendLine($"{setEnvironmentVariable} ROSLYN_TEST_IOPERATION={iop}");
+
+                command.AppendLine($"dotnet {commandLineArguments}");
+
+                // We want to collect any dumps during the post command step here; these commands are ran after the
+                // return value of the main command is captured; a Helix Job is considered to fail if the main command returns a
+                // non-zero error code, and we don't want the cleanup steps to interefere with that. PostCommands exist
+                // precisely to address this problem.
+                var payloadDirectory = Path.Combine(msbuildTestPayloadRoot, Path.GetDirectoryName(assemblyInfo.AssemblyPath)!);
+
+                if (isUnix)
+                {
+                    // Write out this command into a separate file; unfortunately the use of single quotes and ; that is required
+                    // for the command to work causes too much escaping issues in MSBuild.
+                    File.WriteAllText(Path.Combine(payloadDirectory, "copy-dumps.sh"), "find . -name '*.dmp' -exec cp {} $HELIX_DUMP_FOLDER \\;");
+                }
+
+                var postCommand = new StringBuilder();
+                postCommand.AppendLine(isUnix ? "./copy-dumps.sh" : "for /r %%f in (*.dmp) do copy %%f %HELIX_DUMP_FOLDER%");
```

Comment:
nit: could use the branch on line 181 to append this and have an 'else' to append the Windows version of the command. Not very important to me though.

Reply:
Ah, yeah, good idea.